homeassistant/helpers/frame.py

```diff
@@ -85,6 +88,12 @@ def report_integration(
     """
     found_frame, integration, path = integration_frame
 
+    # Keep track of integrations already reported to prevent flooding
+    key = f"{integration}:{what}"
```

Comment:
the key could be adjusted to include the top filename/lineno:
```suggestion
    key = f"{integration}:{what}:{found_frame.filename}:{found_frame.lineno}"
```